server/s3server.cc

```diff
@@ -531,88 +532,66 @@ void fini_auth_ssl() {
 int create_global_index(struct m0_uint128 &root_index_oid,
                         const uint64_t &u_lo_index_offset) {
   int rc;
-  struct m0_op *ops[1] = {NULL};
-  struct m0_op *sync_op = NULL;
-  struct m0_idx idx;
-  unsigned short motr_op_wait_period =
-      g_option_instance->get_motr_op_wait_period();
 
-  memset(&idx, 0, sizeof(idx));
-  ops[0] = NULL;
+  const auto motr_op_wait_period = g_option_instance->get_motr_op_wait_period();
+  unsigned n_retry = g_option_instance->get_max_retry_count() + 1;
+
   // reserving an oid for root index -- M0_ID_APP + offset
+  const auto m0_timeout = m0_time_from_now(motr_op_wait_period, 0);
   init_s3_index_oid(root_index_oid, u_lo_index_offset);
+
+  struct m0_idx idx;
+  memset(&idx, 0, sizeof idx);
+
   m0_idx_init(&idx, &motr_uber_realm, &root_index_oid);
-  m0_entity_create(NULL, &idx.in_entity, &ops[0]);
-  m0_op_launch(ops, 1);
 
-  rc = m0_op_wait(ops[0], M0_BITS(M0_OS_FAILED, M0_OS_STABLE),
-                  m0_time_from_now(motr_op_wait_period, 0));
-  rc = (rc < 0) ? rc : m0_rc(ops[0]);
-  if (rc < 0) {
-    if (rc != -EEXIST) {
-      goto FAIL;
-    }
-  }
-  if (rc != -EEXIST) {
-    rc = m0_sync_op_init(&sync_op);
-    if (rc != 0) {
-      goto FAIL;
+  for (; --n_retry > 0; ::sleep(1)) {  // suspend current thread for 1 second
+    struct m0_op *cr_op = nullptr;
+
+    m0_entity_create(NULL, &idx.in_entity, &cr_op);
+    m0_op_launch(&cr_op, 1);
+
+    rc = m0_op_wait(cr_op, M0_BITS(M0_OS_FAILED, M0_OS_STABLE), m0_timeout);
+    if (rc >= 0) {
+      rc = m0_rc(cr_op);
     }
+    teardown_motr_cancel_wait_op(cr_op, m0_timeout);
```

Comment:
This function should not get called in case if rc returned is EEXIST right. That means global inde table has already being created by some other s2server instance. So we are good there

Reply:
Almost the same was called previously.
  if (rc != -EEXIST) {
    rc = m0_sync_op_init(&sync_op);
    ...
  }
  if (ops[0] != NULL) {
    teardown_motr_op(ops[0]);
  }